typescript/promisify-anything/src/query-athena.ts

```diff
@@ -0,0 +1,49 @@
+import * as restate from "@restatedev/restate-sdk";
+import * as athena from "@aws-sdk/client-athena";
+
+const QUERY_SQL = `
+    SELECT sum("table"."value") AS answer
+    FROM "demo_db"."table"
+`;
+
+export default async function queryAthena(ctx: restate.RpcContext, query?: string) {
+  const requestId = ctx.rand.uuidv4(); // make up a stable idempotency token for calling Athena API
+
+  const client = new athena.AthenaClient({});
+  const executionId = await ctx.sideEffect(async () => {
+    try {
+      const startQueryResult = await client.send(
+        new athena.StartQueryExecutionCommand({
+          QueryString: query ?? QUERY_SQL,
+          WorkGroup: "demo-workgroup",
+          ClientRequestToken: requestId,
+        }),
+      );
+      return startQueryResult.QueryExecutionId as string;
+    } catch (error) {
+      if (error instanceof athena.AthenaServiceException)
+        if (error.$fault === "client") throw new restate.TerminalError(error.message);
+
+      throw error; // rethrow other exceptions -- side effect retry policy will handle them
+    }
+  });
```

Comment:
@tillrohrmann WDYT about this alternative:

```ts
const startQueryOutput = await ctx.sideEffect(
  stopOnError(
    (e) => (e as { $fault?: string }).$fault === "client",
    async () =>
      await client.send(
        new athena.StartQueryExecutionCommand({
          QueryString: query ?? QUERY_SQL,
          WorkGroup: "demo-workgroup",
          ClientRequestToken: requestId,
        }),
      ),
  ),
);
```

Reply:
I think this is a good improvement which makes things easier to understand :-)

Reply:
Done!